gensim/models/word2vec.py

```diff
@@ -1174,18 +1185,30 @@ def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
                     )
                 checked_string_types += 1
             if sentence_no % progress_per == 0:
+
                 logger.info(
                     "PROGRESS: at sentence #%i, processed %i words, keeping %i word types",
-                    sentence_no, total_words, len(vocab)
+                    sentence_no, total_words, len(wrapped_dict)
                 )
-            for word in sentence:
-                vocab[word] += 1
+
             total_words += len(sentence)
+            wrapped_dict.update(sentence)
 
-            if self.max_vocab_size and len(vocab) > self.max_vocab_size:
+            if self.max_vocab_size and len(wrapped_dict) > self.max_vocab_size:
+                vocab = wrapped_dict.get()
```

Comment:
This looks very, very wrong. It's performing the exact same crude, slow premature pruning-to-save-memory when using a Bounter as when using a dictionary. And that pruning requires converting the whole Bounter to a memory-consuming regular dict each prune. And then the results of each costly prune are ignored, because nothing's done with the 'vocab' local. So this seems to break `max_vocab_size` for the old case, while retaining it as the source of a time- and space-consuming no-op in the bounter case. To truly test/evaluate a change here will require some testing (ad hoc or unit) of cases where `max_vocab_size` is operative, including timing & peak memory effects of the change vs original implementation.

Reply:
Ok
I will think about this and work on it.